python_ta/reporters/html_reporter.py

```diff
@@ -1,31 +1,99 @@
 import os
 
 from jinja2 import Environment, FileSystemLoader
+from datetime import datetime
 
-from .plain_reporter import PlainReporter
+from .color_reporter import ColorReporter
 
 THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+_SPACES = '&nbsp;&nbsp;&nbsp;&nbsp;'  # 4 HTML non-breaking spaces
 
 
-class HTMLReporter(PlainReporter):
+class HTMLReporter(ColorReporter):
     # Override this method
     def print_messages(self, level='all'):
         # Sort the messages.
         self.sort_messages()
+        # Call these two just to fill snippet attribute of Messages
+        # (and also to fix weird bad-whitespace thing):
+        self._colour_messages_by_type(style=False)
+        self._colour_messages_by_type(style=True)
 
         template = Environment(loader=FileSystemLoader(THIS_DIR)).get_template('templates/template.txt')
+        output_path = THIS_DIR + '/templates/output.html'
 
-        with open('output.html', 'w') as f:
-            for msg in self._error_messages:
-                msg_new = msg.msg.replace('\r', '')
-                msg_new = msg_new.replace('\n', '<br/>&emsp;&emsp;')
-                i = self._error_messages.index(msg)
-                self._error_messages[i] = msg._replace(msg=msg_new)
+        # Date/time (24 hour time) format:
+        # Generated: ShortDay. ShortMonth. PaddedDay LongYear, Hour:Min:Sec
+        dt = "Generated: " + str(datetime.now().
+                                 strftime("%a. %b. %d %Y, %H:%M:%S"))
+        with open(output_path, 'w') as f:
+            f.write(template.render(date_time=dt,
+                                    mod_name=self._module_name,
+                                    code=self._sorted_error_messages,
+                                    style=self._sorted_style_messages))
+        print("HTML Python TA report created. Please see output.html "
+              "within pyta/python_ta/reporters/templates.")
 
-            for msg in self._style_messages:
-                msg_new = msg.msg.replace('\r', '')
-                msg_new = msg_new.replace('\n', '<br/>&emsp;&emsp;')
-                i = self._style_messages.index(msg)
-                self._style_messages[i] = msg._replace(msg=msg_new)
+    def _add_line(self, msg, n, linetype):
```

Comment:
Fixed, with some alterations: 
* `_space` is an instance variable, and can be set in constructors. 
* Alongside `_space` is another instance variable: the dict `_colouring`, which holds what colour class names mean for each reporter. 
    * This addition means that _both_ `_colourify` and `_add_line` no longer need to be overridden in `HTMLReporter`, because both of them have been changed to accommodate the use of `_colouring`.